remote-api/src/main/java/org/alfresco/rest/api/impl/AspectsImpl.java

```diff
@@ -0,0 +1,321 @@
+    {
+        this.classDefinitionMapper = classDefinitionMapper;
+    }
+
+    public void init()
+    {
+        PropertyCheck.mandatory(this, "dictionaryService", dictionaryService);
+        PropertyCheck.mandatory(this, "namespaceService", namespaceService);
+        PropertyCheck.mandatory(this, "classDefinitionMapper", classDefinitionMapper);
+    }
+
+
+    @Override
+    public CollectionWithPagingInfo<Aspect> listAspects(Parameters params)
+    {
+        Paging paging = params.getPaging();
+        AspectsFilter query = getQuery(params.getQuery());
+        Stream<QName> aspectList = null;
+
+        if(query != null && query.getModelIds() != null)
+        {
+            aspectList = query.getModelIds().parallelStream().map(this::getModelAspects).flatMap(Collection::parallelStream);
+        }
+        else if(query != null && query.getParentIds() != null)
+        {
+            aspectList = query.getParentIds().parallelStream().map(this::getChildAspects).flatMap(Collection::parallelStream);
+        }
+        else
+        {
+            aspectList = this.dictionaryService.getAllAspects().parallelStream();
+        }
+
+        List<Aspect> allAspects = aspectList.filter((qName) -> filterAspect(query, qName))
+                .map((qName) -> this.convertToAspect(dictionaryService.getAspect(qName)))
+                .collect(Collectors.toList());
+        return createPagedResult(allAspects, paging);
+    }
+
+    @Override
+    public Aspect getAspectById(String aspectId)
+    {
+        if(aspectId == null)
+            throw new InvalidArgumentException("aspectId is null");
+
+        AspectDefinition aspectDefinition = null;
+
+        try
+        {
+           aspectDefinition = dictionaryService.getAspect(QName.createQName(aspectId, this.namespaceService));
+        }
+        catch (NamespaceException exception)
+        {
+            throw new InvalidArgumentException(exception.getMessage());
+        }
+
+        return this.convertToAspect(aspectDefinition);
+    }
+
+    public Aspect convertToAspect(AspectDefinition aspectDefinition)
+    {
+        List<PropertyDefinition> properties = this.classDefinitionMapper.fromDictionaryClassDefinition(aspectDefinition, dictionaryService).getProperties();
+        return new Aspect(aspectDefinition, dictionaryService, properties);
+    }
+
+    public AspectsFilter getQuery(Query queryParameters)
+    {
+        if (queryParameters != null)
+        {
+            AspectQueryWalker propertyWalker = new AspectQueryWalker();
+            QueryHelper.walk(queryParameters, propertyWalker);
+            return new AspectsFilter(propertyWalker.getModelIds(), propertyWalker.getParentIds(), propertyWalker.getMatchedPrefix(), propertyWalker.getNotMatchedPrefix());
+        }
+        return null;
+    }
+
+    private Collection<QName> getModelAspects(String modelId) {
+        ModelDefinition modelDefinition =  null;
+
+        if(modelId == null)
```

Comment:
Please add a space between the if and opening bracket to be consistent with coding standards for this project.